04_pig_latin/pig_latin.rb

```diff
@@ -1 +1,46 @@
 #write your code here
+def translate(s)
+	w = s.clone # w = words
+
+	#If there are more than 1 word 
+	if w.index(" ")
+		w = w.split(" ")
+		w.map! { |word| word = translate(word)}
+		w.join(" ")
+
+	#Add -ay to end
+	elsif isVowel(w[0])
+		w << "ay"
+
+	#If 3 consonants, shift to end and add "ay"
+	elsif isConsonants(w[0..2])
+		w = w.split("")
+		c = w.shift << w.shift << w.shift
+		w << c << "ay"
+		w.join
+
+	##If 2 consonants, shift to end and add "ay"
+	elsif isConsonants(w[0..1])
+		w = w.split("")
+		c = w.shift << w.shift
+		w << c << "ay"
+		w.join
+
+	#If 1 consonant, shift to end and add "ay"
+	elsif isConsonants(w[0])
+		w = w.split("")
+		c = w.shift
+		w << c << "ay"
+		w.join
+	end
+end
+
+def isVowel(c)
+	c.index(/[aeiouAEIOU]/) == nil ? false : true
+end
+
+def isConsonants(chars)
+	chars.gsub!("qu", "qq") if chars.include?("qu")
+	chars.split("").each { |c| return false if isVowel(c) }
+	true
+end
```

Comment:
I tested this function by calling `translate("here is quit place")` and it gave me an error. There're few changes can be implemented to may this code work :
1. here you're passing argument "s" and then you clone it but don't use it. You can pass argument and use it in your code.
2. you can split your incoming string, it creates an array of words. 
3. then you can iterate through that array checking each word, modify it and add to a new array that can store modified words.
4. at the end you can join the array of strings  and return it.